core/rawdb/accessors_chain.go

```diff
@@ -694,8 +694,21 @@ type receiptLogs struct {
 
 // DecodeRLP implements rlp.Decoder.
 func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
+	// Retrieve the entire receipt blob as we need to try multiple decoders
+	blob, err := s.Raw()
+	if err != nil {
+		return err
+	}
+	// Check to see if this is a celo dynamic fee receipt.
+	if types.IsCeloDynamicFeeReceipt(blob) {
+		var stored types.CeloDynamicFeeStoredReceiptRLP
+		err := rlp.DecodeBytes(blob, &stored)
+		r.Logs = stored.Logs
```

Comment:
Do we want to assign these to r.Logs if err might not be nil?

Reply:
It doesn't matter if we assign them or not, if the error returned is non nil then rlp.Decode isn't making any guarantee about the state of the variable you are decoding into. Meaning if the error is non nil don't rely on the value.

Reply:
Ok, If it doesn't matter, why not stay in the good practices?
It's not the case where you are returning an error plus some variable that lives in the function, you are assigning something (as you say, without any guarantee about the state of the variable you are decoding into), to the actual object.
Probably works and has the same output, but then you have to take in count a few things.
* what happens if a newer version changes the DecodeBytes and now the stored object has something weird after sending an error
* what will happen to a dev checking this function in the future? Why the dev decided to do something "smelly"? It was part of a fix, or is this a bug?
* what would happen if the one using this function checks other thing (you are heavily relaying that after an error, nobody is going to use THAT object, that could be corrupted) 

To resume, haha. I think that if it doesn't matter, you should stick to good practices.

Reply:
I don't see this as a bad practice, it just differs a bit from what people are used to seeing and is more concise.

I guess my point is that however DecodeBytes or whatever function is modified in the future, in the end a value and an error will be delivered to some caller. If the error is non nil then the caller should not rely on the value, and yes we heavily rely on that, but the entire go language relies on that, if we don't want to rely on that then we shouldn't use go.

It is true however that this more concise approach has caused a lot of confusion here, so I will accept your suggestion.